app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -122,6 +121,42 @@ def bulk_resume
         }
       end
 
+      api :POST, '/tasks/bulk_cancel', N_('Cancel all cancellable tasks')
+      param :search, String, :desc => N_('Cancel tasks matching search string')
+      param :task_ids, Array, :desc => N_('Cancel specific tasks by ID')
+      def bulk_cancel
+        scope = resource_scope
+        scope = scope.search_for(params[:search]) if params[:search]
+        scope = scope.select('DISTINCT foreman_tasks_tasks.*')
+        if params[:search].nil? && params[:task_ids].nil?
+          raise ArgumentError, _('Please provide at least one of search or task_ids parameters in the request')
+        end
+        scope = scope.where(:id => params[:task_ids]) if params[:task_ids]
+
+        cancelled = []
+        failed = []
+        skipped = []
+        scope.each do |task|
+          if task.cancellable?
+            begin
+              task.cancel
+              cancelled << task_hash(task)
+            rescue RuntimeError
```

Comment:
An exception should be raised only if the task's execution plan cannot be loaded. However all tasks which are `cancellable?` should have an execution plan. That would mean we can drop the `begin-rescue-end` block

Reply:
Does this mean task canceling can't fail?

Reply:
If a task is `cancellable?` then the cancelling should not raise, unless something happened between calls to `cancellable?` and `cancel`.

Under normal conditions, if a task is `cancellable?` then the worst thing that can happen is that the cancel won't actually do anything.